### DISCUSSION TITLE
hot reload issue and pinia store misload problem

### DISCUSSION THREAD

**Discussion by han-ysys** [quasarframework/quasar#15691] [General - Components / Directives / etc] [answered]
### What happened?

I've created 2 pinia stores, one named 'useSampleInfoStore' and another named 'useReadsStore'. 
```typecript
const useSampleInfoStore = defineStore('table', {
  state: () => ({
    sampleInfo: [] as SampleInfo[],
  }),
  actions: {
    addSample(sample: Sample) {
      this.sampleInfo.push({
        ...sample,
        targetSeq: '',
        reads: [] as SampleInfo['reads'],
      });
    },
    removeSample(sample: Sample) {
      try {
        const index = this.sampleInfo.indexOf(
          this.sampleInfo.filter((s) => s.name === sample.name)[0]
        );
        if (index > -1) {
          this.sampleInfo.splice(index, 1);
        }
      } catch (e) {
        console.error(e);
      }
    },
    targetSeq(seq: Sequence) {
      this.sampleInfo.forEach((sample) => {
        if (sample.name === seq.header) {
          sample.targetSeq = seq.seq;
        }
      });
    },
  },
});

const useReadsStore = defineStore('table', {
  state: () => ({
    reads: { '': 0 } as { [key: string]: number },
  }),
  actions: {
    addSeq(read: string) {
      this.reads[read] = (this.reads[read] || 0) + 1;
    },
    setSeq(reads: { [key: string]: number }) {
      this.reads = reads;
    },
  },
  getters: {
    count(state) {
      return Object.values(state.reads).reduce((a, b) => a + b, 0);
    },
  },
});
```
I got a `sampleInfo.addSample is not a function` error using the sampleInfoStore in a component. Tried to log the object, but surprisingly find out that the `useSampleInfoStore()` I logged to the console is actually a `useReadsStore()` object. I am not even importing `useReadsStore()` in the component. 
```typescript
<script lang="ts" setup>
import { useSampleInfoStore } from '../../stores/sampleinfo';
const onAdd = () => {
  const sampleInfo = useSampleInfoStore();
  const clone: Sample = JSON.parse(JSON.stringify(formInline));
  if ((clone.group = '')) clone.group = 'None';
  console.log(
    sampleInfo,
    typeof sampleInfo.addSample,
    typeof sampleInfo.addSeq
  );
  sampleInfo.addSample(clone);
  resetForm();
};
</script>
```
<img width="811" alt="截屏2023-04-08 10 57 53" src="https://user-images.githubusercontent.com/70315648/230700341-22e0426e-a557-4ca0-aacc-2a81dd9c6824.png">

My IDE can tell that 'addSeq' is not a function of sampleInfo.
<img width="805" alt="截屏2023-04-08 10 55 59" src="https://user-images.githubusercontent.com/70315648/230700294-8f50a2b9-6474-4274-bd65-e8e31598e920.png">
I think it is a quasar problem because quasar is not hot reloading either on `quasar dev` or `quasar dev -m electron`. The only error I saw at the compiling stage is `[1397:0408/105231.985700:ERROR:iopm_power_source_sampling_event_source.cc(31)] IOPMPowerSource service not found`. I've got no clue about that. 

### What did you expect to happen?

quasar hot reload works well and the weird store issue is fixed.

### Reproduction URL

https://github.com/han-ysys/crisprstitch

### How to reproduce?

Sorry about the Github repo URL. The environment was a little bit complicated.
To reproduce, 
1. clone the GitHub repo and `npm install`.
2. `quasar dev` and click the 'Start' button on the jumped-out page. 
3. click 'example' and then 'add'.
4. It results in a weird log in the console.

### Flavour

Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)

### Areas

Components (quasar)

### Platforms/Browsers

Chrome, Electron

### Quasar info output

```Shell
Operating System - Darwin(21.2.0) - darwin/x64
NodeJs - 16.15.0

Global packages
  NPM - 8.5.5
  yarn - 1.22.10
  @quasar/cli - 2.0.0
  @quasar/icongenie - Not installed
  cordova - Not installed

Important local packages
  quasar - 2.11.8 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app-vite - 1.2.1 -- Quasar Framework App CLI with Vite
  @quasar/extras - 1.15.11 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.2.47 -- The progressive JavaScript framework for building modern web UI.
  vue-router - 4.1.6
  pinia - 2.0.33 -- Intuitive, type safe and flexible Store for Vue
  vuex - Not installed
  vite - 2.9.15 -- Native-ESM powered web dev build tool
  eslint - 8.36.0 -- An AST-based pattern checker for JavaScript.
  electron - 23.2.0 -- Build cross platform desktop apps with JavaScript, HTML, and CSS
  electron-packager - Not installed
  electron-builder - Not installed
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed

Quasar App Extensions
  *None installed*

Networking
  Host - iMac.local
  en0 - 192.168.127.17
  en1 - 192.168.127.178
```

### Relevant log output

```Shell
[5610:0408/122237.361934:ERROR:iopm_power_source_sampling_event_source.cc(31)] IOPMPowerSource service not found
Debugger listening on ws://127.0.0.1:5858/376ee7b8-25e2-46ca-ab11-ea5eef3d7dfb
```

### Additional context

The component where I am using pinia store is 'src/components/sampleInfo/SampleForm.vue'

**Comment by pdanpdan** [comment]
I can find no `addSeq` when searching the repo

**Reply by han-ysys** [reply]
addSeq is one of the actions of useReadsStore() in src/stores/reads.ts

**Comment by rstoenescu** [comment, marked as answer]
Looking only at your copy-pasted code I see that you are defining both stores with the same name ("table"). Use different names for each.

```
defineStore('table'
```
